### PR TITLE
redis_info module - adding user auth and section option

### PR DESCRIPTION
**Pull request by vicmunoz**
##### SUMMARY

Module redis_info minor changes. Adding 'login_user' parameter for authentication. Adding 'section' parameter to retrieve specific info section

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
redis_info

### CODE REVIEW

File: changelogs/fragments/6643-redis_info_auth.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - redis_info - adding user auth and section option
```
Comment (felixfontein) [inline changelogs/fragments/6643-redis_info_auth.yml:3]:
```suggestion
  - redis_info - adding user authentication and ``section`` options (https://github.com/ansible-collections/community.general/pull/6643).
```

File: plugins/modules/redis_info.py [hunk 2]
```diff
@@ -193,46 +213,41 @@
...
+else:
+    redis_found = True
 
-from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.basic import AnsibleModule
 from ansible.module_utils.common.text.converters import to_native
-
-
-def redis_client(**client_params):
-    return StrictRedis(**client_params)
```
Comment (felixfontein) [inline plugins/modules/redis_info.py:207]: This helper function was likely needed by the unit tests, which now fail. You need to fix them.

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments.

Line 63: please remove the `requirements` key. It is no longer necessary due to the redis docs_fragment one.

File: plugins/modules/redis_info.py (hunk 2 above)
Comment (vicmunoz) [reply plugins/modules/redis_info.py:207]: tests are fixed in last commit

**Comment by vicmunoz** [comment]
@felixfontein I've fixed all failed unit and sanity tests I've found. Please let me know if I'm missing something or it can be merged. Thanks!

File: plugins/modules/redis_info.py [hunk 3]
```diff
@@ -2,49 +2,69 @@
...
+        default: localhost
+    login_port:
+        description:
+            - The port to connect to.
+        type: int
+        default: 6379
+    login_password:
+        description:
+            - The password used to authenticate with, when authentication is enabled for the Redis server.
+        type: str
```
Comment (felixfontein) [inline plugins/modules/redis_info.py:39]: These are provided by the docs fragment, there's no need to replicate them (with slightly different descriptions).
```suggestion
```

File: plugins/modules/redis_info.py [hunk 4]
```diff
@@ -2,49 +2,69 @@
...
+# GNU General Public License v3.0+
+# (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
 
-DOCUMENTATION = r'''
+DOCUMENTATION = '''
```
Comment (felixfontein) [inline plugins/modules/redis_info.py:14]: Please keep the `r` for raw strings (same for the other two you removed).
```suggestion
DOCUMENTATION = r'''
```

File: plugins/modules/redis_info.py [hunk 5]
```diff
@@ -2,49 +2,69 @@
...
   install it with pip (C(pip install redis)) or with a package manager.
   U(https://github.com/andymccurdy/redis-py)
 seealso:
 - module: community.general.redis
-requirements: [ redis ]
 author: "Pavlo Bashynskyi (@levonet)"
 '''
 
-EXAMPLES = r'''
+EXAMPLES = '''
```
Comment (felixfontein) [inline plugins/modules/redis_info.py:67]:
```suggestion
EXAMPLES = r'''
```

File: plugins/modules/redis_info.py [hunk 6]
```diff
@@ -54,7 +74,7 @@
     var: result.info
 '''
 
-RETURN = r'''
+RETURN = '''
```
Comment (felixfontein) [inline plugins/modules/redis_info.py:77]:
```suggestion
RETURN = r'''
```

File: plugins/modules/redis_info.py [hunk 7]
```diff
@@ -188,51 +208,33 @@
...
+    module_args.update(redis_auth_args)
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+    fail_imports(module, module.params['tls'])
+    section = module.params['section']
 
     # Connect and check
-    client = redis_client(host=login_host, port=login_port, password=login_password)
+    client = RedisAnsible(module)
     try:
-        client.ping()
```
Comment (felixfontein) [inline plugins/modules/redis_info.py:231]: Is there a reason that `client.ping()` is no longer called? This is the main change in behavior.

File: plugins/modules/redis_info.py [hunk 8]
```diff
@@ -188,51 +208,33 @@
...
-    except Exception as e:
-        module.fail_json(msg="unable to connect to database: %s" % to_native(e), exception=traceback.format_exc())
-
-    info = client.info()
+        info = client.connection.info(section=section)
```
Comment (felixfontein) [inline plugins/modules/redis_info.py:234]: Is `section='default'` the default value for `info()`? (Also has it been that for all supported versions of the redis library?)

**Comment by felixfontein** [comment]
Also please notice that the unit tests fail on Python 2.6 (due to an older redis version).

**Comment by felixfontein** [comment]
needs_info

**Pull request closed without merging**